### PR TITLE
Re-implement Gpio, I2c, Pwm, Udp modules in C

### PR DESCRIPTION
**Pull request by yichoi**
Related issue: https://github.com/Samsung/iotjs/issues/439
This commit converts every remaining C++ code into C.

### CODE REVIEW

File: src/platform/arm-nuttx/iotjs_module_i2c-arm-nuttx.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+#if defined(__NUTTX__)
+
+
+#include "module/iotjs_module_i2c.h"
+
+void I2cSetAddress(uint8_t address) {
+  IOTJS_ASSERT(!"Not implemented");
+}
+void ScanWorker(uv_work_t* work_req) {
```
Comment (LaszloLango) [inline src/platform/arm-nuttx/iotjs_module_i2c-arm-nuttx.c]: minor style issue: missing spaces between functions

File: src/platform/arm-nuttx/iotjs_module_pwm-arm-nuttx.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+#include "iotjs_def.h"
+#include "module/iotjs_module_pwm.h"
+
+int PwmInitializePwmPath(iotjs_pwmreqdata_t* req_data) {
+  IOTJS_ASSERT(!"Not implemented");
+  return 0;
+}
+void ExportWorker(uv_work_t* work_req) {
```
Comment (LaszloLango) [inline src/platform/arm-nuttx/iotjs_module_pwm-arm-nuttx.c]: ditto

File: src/module/iotjs_module_i2c.c [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+ */
+
+
+#include "iotjs_def.h"
+#include "iotjs_objectwrap.h"
+#include "iotjs_module_i2c.h"
+
+
+#define THIS iotjs_i2creqwrap_t* i2creqwrap
+void iotjs_i2creqwrap_initialize(THIS, const iotjs_jval_t* jcallback,
```
Comment (nova0821) [inline src/module/iotjs_module_i2c.c]: Please insert one or two blank lines between ``define`` and function declaration.

File: src/module/iotjs_module_i2c.c [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+  iotjs_jhandler_return_null(jhandler);
+}
+
+
+JHANDLER_FUNCTION(ReadBlock) {
+  JHANDLER_CHECK_ARGS(4, number, number, number, function);
+
+  const iotjs_jval_t* jcallback = JHANDLER_GET_ARG(0, function);
+
+  iotjs_i2creqwrap_t* req_wrap = IOTJS_ALLOC(iotjs_i2creqwrap_t);
```
Comment (nova0821) [inline src/module/iotjs_module_i2c.c]: Please change ``0`` to ``3``, it looks typo error.

File: src/module/iotjs_module_i2c.c [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+
+#define THIS iotjs_i2creqwrap_t* i2creqwrap
+
+
+void iotjs_i2creqwrap_initialize(THIS, const iotjs_jval_t* jcallback,
+                                 I2cOp op) {
+  IOTJS_VALIDATED_STRUCT_CONSTRUCTOR(iotjs_i2creqwrap_t, i2creqwrap);
+  iotjs_reqwrap_initialize(&_this->reqwrap, jcallback, (uv_req_t*)&_this->req,
+                           (void*)i2creqwrap);
+
```
Comment (nova0821) [inline src/module/iotjs_module_i2c.c]: Need to assign ``op`` to ``_this->req_data.op``.

File: src/module/iotjs_module_pwm.c [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+
+#include "iotjs_def.h"
+#include "iotjs_objectwrap.h"
+#include "iotjs_module_pwm.h"
+
+
+#define THIS iotjs_pwmreqwrap_t* pwmreqwrap
+
+void iotjs_pwmreqwrap_initialize(THIS, const iotjs_jval_t* jcallback,
+                                  PwmOp op) {
```
Comment (hs0225) [inline src/module/iotjs_module_pwm.c]: Please check code style.

**Comment by hs0225** [comment]
LGTM

**Pull request merged**